src/transformers/models/mobilevit/feature_extraction_mobilevit.py

```diff
@@ -0,0 +1,205 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Feature extractor class for MobileViT."""
+
+from typing import Optional, Union
+
+import numpy as np
+from PIL import Image
+
+from ...feature_extraction_utils import BatchFeature, FeatureExtractionMixin
+from ...image_utils import ImageFeatureExtractionMixin, ImageInput, is_torch_tensor
+from ...utils import TensorType, logging
+
+
+logger = logging.get_logger(__name__)
+
+
+class MobileViTFeatureExtractor(FeatureExtractionMixin, ImageFeatureExtractionMixin):
+    r"""
+    Constructs a MobileViT feature extractor.
+
+    This feature extractor inherits from [`FeatureExtractionMixin`] which contains most of the main methods. Users
+    should refer to this superclass for more information regarding those methods.
+
+    Args:
+        do_resize (`bool`, *optional*, defaults to `True`):
+            Whether to resize the input to a certain `size`.
+        size (`int` or `Tuple(int)`, *optional*, defaults to 288):
+            Resize the input to the given size. If a tuple is provided, it should be (width, height). If only an
+            integer is provided, then the input will be resized to (size, size). Only has an effect if `do_resize` is
```

Comment:
Given the code of resize below, it seems that,  in case you provide an integer, then the shorter side of the image is matched to this number.

Also, this is implemented in image_utils.py (you can specify `default_to_square=False`), wondering if we need to duplicate the resize method here.

Reply:
I copied this from CLIP and didn't really pay attention to the doc string. If it's wrong it should also be fixed there.

However, the reason for not using the method from image_utils.py is that that method fails when the input to the feature extractor is not a PIL Image, and this one doesn't. I can only assume this is why CLIP wrote their own.

Reply:
I'll post a reproducible case in an issue. (I agree that it would be better to reuse the existing code from image_utils.)

Reply:
It turns out that the issue I had was not in resize, so I'll change this to use the one from image_utils.